tests/testthat/test-data_select.R

```diff
@@ -402,3 +402,19 @@ test_that("old solution still works", {
     c("Sepal.Length", "Sepal.Width")
   )
 })
+
+test_that("data_select renames variables on the fly", {
+  data(mtcars)
+  out <- data_select(mtcars, c(new = "mpg", old = "cyl", hoho = "wt"))
+  expect_named(out, c("new", "old", "hoho"))
+  data_select(mtcars, c(new = "mpg", "cyl", hoho = "wt"))
```

Comment:
Missing `out <-` in several places I think.

Also maybe you can make it more readable by removing the assignment and just having:
```r
expect_named(
  data_select(mtcars, c(new = "mpg", old = "cyl", hoho = "wt")),
  c("new", "old", "hoho")
)
...
```